### PR TITLE
Update Astro docs to use v0.21

### PR DESCRIPTION
**Pull request by jonathantneal**
## Changes

This updates the Astro docs site to use Astro v0.21.

## Testing

docs site only

## Docs

docs site only

### CODE REVIEW

**Comment by jonathantneal** [comment]
The issue may be in how Astro is handling Vite aliases.

cc @natemoo-re

**Example**
```astro
<!-- src/pages/index.astro -->
<script type="module" src={Astro.resolve('../scripts/script-a.js')} />
```
```js
// src/scripts/script-a.js
import fn from '~/components/script-b.js'

fn()
```
```js
// src/scripts/script-b.js
export default function Fn() {
	document.title = 'Component loaded.'
}
```

```
400 http://localhost:3000/@id/src/components/script-b.js net::ERR_ABORTED 404 (Not Found)
```

https://stackblitz.com/edit/github-hp5d6g?file=src%2Fpages%2Findex.astro&on=stackblitz

File: docs/snowpack.config.mjs [hunk 1]
```diff
@@ -1,8 +0,0 @@
-export default {
```
Comment (drwpow) [inline docs/snowpack.config.mjs:1]: 🥲

File: docs/astro.config.mjs [hunk 2]
```diff
@@ -8,4 +9,12 @@ export default {
     // Needed for Algolia search component
     '@astrojs/renderer-react',
   ],
-};
+  vite: {
+    resolve: {
+      alias: {
+        '~': './src',
+        'components': './src/components',
```
Comment (drwpow) [inline docs/astro.config.mjs:16]: This is a soft suggestion: aliasing directories was required in Snowpack, but is not in Vite. I think we’d prefer people use “real” paths as the default. But I’d be in favor of moving these to a guide on aliasing for people who want to opt in.

Comment (drwpow) [reply docs/astro.config.mjs:16]: Only coming from past experience where many people would set up aliasing, only to find them confusing to work with

File: docs/src/components/Header/Header.astro [hunk 3]
```diff
@@ -1,8 +1,8 @@
 ---
 import SkipToContent from './SkipToContent.astro';
 import SidebarToggle from './SidebarToggle.tsx';
-import LanguageSelect from './LanguageSelect.jsx';
-import Search from "./Search.jsx";
```
Comment (drwpow) [inline docs/src/components/Header/Header.astro:5]: 👍🏻

File: docs/astro.config.mjs (hunk 2 above)
Comment (matthewp) [reply docs/astro.config.mjs:16]: I think the purpose here is just to test that the feature works, since we do support it.

Comment (jonathantneal) [reply docs/astro.config.mjs]: 1. This aliasing has been updated in https://github.com/snowpackjs/astro/pull/1738/commits/b37e2dd8b697181f28c42754c9e69bc29de7b59e
2. This aliasing may be superseded by https://github.com/snowpackjs/astro/pull/1747 to provide consistent aliasing to both Astro and the IDE. IMHO, aliasing that is not also picked up by the editor is confusing to work with.

**Comment by drwpow** [comment]
Is the PR description up-to-date? Is there anything still blocking this (the SVG issue has been fixed in the compiler right)?

**Review by jasikpark — APPROVED** [review]
lgtm! I'm excited to get the docs & www working w/ v0.21

**Pull request merged**